tsl/src/chunk_api.c

```diff
@@ -332,6 +332,17 @@ get_hypercube_from_slices(Jsonb *slices, const Hypertable *ht)
 	return hc;
 }
 
+/**
+ * Get the chunk status.
+ */
+Datum
+chunk_status(PG_FUNCTION_ARGS)
```

Comment:
we can eliminate this function by querying chunk table using SQL in the procedure.

Reply:
Any metadata changes will need changes to views that use chunks. But either way is fine with me.